workbench-core/accounts/src/models/accounts/listAccountsRequest.test.ts

```diff
@@ -0,0 +1,66 @@
+import JSONValue from '@aws/workbench-core-base/lib/types/json';
+import { ListAccountsRequestParser } from './listAccountsRequest';
+
+describe('ListAccountsRequestParser', () => {
+  let requestObject: Record<string, JSONValue>;
+
+  describe('when pageSize', () => {
+    let expectedPageSize: number;
+
+    describe('is valid', () => {
+      beforeEach(() => {
+        expectedPageSize = 5;
+        requestObject = {
+          pageSize: `${expectedPageSize}`
+        };
+      });
+
+      test('it parses the page size into a number', () => {
+        const parsed = ListAccountsRequestParser.safeParse(requestObject);
+        expect(parsed.success).toEqual(true);
+        if (parsed.success) {
+          expect(parsed.data.pageSize).toEqual(expectedPageSize);
+        }
+      });
+    });
+
+    describe('is NaN', () => {
+      beforeEach(() => {
+        requestObject = {
+          pageSize: 'nonNumber'
+        };
+      });
+
+      test('it throws an error', () => {
+        const parsed = ListAccountsRequestParser.safeParse(requestObject);
+        expect(parsed.success).toEqual(false);
+      });
+    });
+
+    describe('is greater than 100', () => {
+      beforeEach(() => {
+        requestObject = {
+          pageSize: '101'
+        };
+      });
+
+      test('it throws an error', () => {
+        const parsed = ListAccountsRequestParser.safeParse(requestObject);
+        expect(parsed.success).toEqual(false);
+      });
+    });
+
+    describe('is less than 1', () => {
+      beforeEach(() => {
+        requestObject = {
+          pageSize: '0'
+        };
+      });
+
+      test('it throws an error', () => {
+        const parsed = ListAccountsRequestParser.safeParse(requestObject);
+        expect(parsed.success).toEqual(false);
```

Comment:
I forgot to update the name of the test, which may be where the confusion is. But if you're still expecting that this would throw a Boom message, no that should not be thrown here. We're using `safeParse`, which returns a parsed result that either has the data if the parsing was successful, or a set of issues if the parsing was a failure. If you're asking why we aren't validating the error message coming back on the issue, I didn't add that but will update it to be present.